pkg/cloudprovider/azure/object_store.go

```diff
@@ -18,121 +18,26 @@ package azure
 )
 
 const (
 	storageAccountConfigKey = "storageAccount"
-)
-
-type containerGetter interface {
-	getContainer(bucket string) (container, error)
-}
-
-type azureContainerGetter struct {
-	blobService *storage.BlobStorageClient
-}
-
-func (cg *azureContainerGetter) getContainer(bucket string) (container, error) {
-	container := cg.blobService.GetContainerReference(bucket)
-	if container == nil {
-		return nil, errors.Errorf("unable to get container reference for bucket %v", bucket)
-	}
-
-	return &azureContainer{
-		container: container,
-	}, nil
-}
-
-type container interface {
-	ListBlobs(params storage.ListBlobsParameters) (storage.BlobListResponse, error)
-}
-
-type azureContainer struct {
-	container *storage.Container
-}
-
-func (c *azureContainer) ListBlobs(params storage.ListBlobsParameters) (storage.BlobListResponse, error) {
-	return c.container.ListBlobs(params)
-}
-
-type blobGetter interface {
-	getBlob(bucket, key string) (blob, error)
-}
-
-type azureBlobGetter struct {
-	blobService *storage.BlobStorageClient
-}
-
-func (bg *azureBlobGetter) getBlob(bucket, key string) (blob, error) {
-	container := bg.blobService.GetContainerReference(bucket)
-	if container == nil {
-		return nil, errors.Errorf("unable to get container reference for bucket %v", bucket)
-	}
-
-	blob := container.GetBlobReference(key)
-	if blob == nil {
-		return nil, errors.Errorf("unable to get blob reference for key %v", key)
-	}
+	tenantIDEnvVar          = "AZURE_TENANT_ID"
+	subscriptionIDEnvVar    = "AZURE_SUBSCRIPTION_ID"
+	clientIDEnvVar          = "AZURE_CLIENT_ID"
+	clientSecretEnvVar      = "AZURE_CLIENT_SECRET"
 
-	return &azureBlob{
-		blob: blob,
-	}, nil
-}
-
-type blob interface {
-	CreateBlockBlobFromReader(blob io.Reader, options *storage.PutBlobOptions) error
-	Exists() (bool, error)
-	Get(options *storage.GetBlobOptions) (io.ReadCloser, error)
-	Delete(options *storage.DeleteBlobOptions) error
-	GetSASURI(options *storage.BlobSASOptions) (string, error)
-}
-
-type azureBlob struct {
-	blob *storage.Blob
-}
-
-func (b *azureBlob) CreateBlockBlobFromReader(blob io.Reader, options *storage.PutBlobOptions) error {
-	return b.blob.CreateBlockBlobFromReader(blob, options)
-}
-
-func (b *azureBlob) Exists() (bool, error) {
-	return b.blob.Exists()
-}
-
-func (b *azureBlob) Get(options *storage.GetBlobOptions) (io.ReadCloser, error) {
-	return b.blob.Get(options)
-}
-
-func (b *azureBlob) Delete(options *storage.DeleteBlobOptions) error {
-	return b.blob.Delete(options)
-}
-
-func (b *azureBlob) GetSASURI(options *storage.BlobSASOptions) (string, error) {
-	return b.blob.GetSASURI(*options)
-}
-
-type ObjectStore struct {
-	containerGetter containerGetter
-	blobGetter      blobGetter
-	log             logrus.FieldLogger
-}
-
-func NewObjectStore(logger logrus.FieldLogger) *ObjectStore {
-	return &ObjectStore{log: logger}
-}
+	resourceGroupConfigKey = "resourceGroup"
+)
 
 func getStorageAccountKey(config map[string]string) (string, error) {
```

Comment:
it looks like this function is out-of-date - doesn't include the recent changes to add support for Azure china/german clouds. Let's get it rebased so we can retain support for that here.